### PR TITLE
Allow validation annotations in more places

### PR DESCRIPTION
**Pull request by findepi**

### CODE REVIEW

File: src/test/java/io/airlift/units/TestDataSizeValidator.java [hunk 1]
```diff
@@ -79,57 +79,48 @@ public void testAllowsNullMaxAnnotation()
...
     {
-        ConstrainedDataSize object = new ConstrainedDataSize(new DataSize(11, MEGABYTE));
-        Set<ConstraintViolation<ConstrainedDataSize>> violations = VALIDATOR.validate(object);
+        Set<? extends ConstraintViolation<?>> violations = VALIDATOR.validate(new ConstrainedDataSize(new DataSize(11, MEGABYTE)));
         assertThat(violations).hasSize(2);
-
-        for (ConstraintViolation<ConstrainedDataSize> violation : violations) {
-            assertThat(violation.getConstraintDescriptor().getAnnotation()).isInstanceOf(MaxDataSize.class);
-        }
+        assertThat(violations.stream().map(violation -> violation.getConstraintDescriptor().getAnnotation()))
```
Comment (electrum) [inline src/test/java/io/airlift/units/TestDataSizeValidator.java]: FYI, there is an `extracting` method, though being able to use a stream is nice (I didn't know that worked). This lets us do
```java
assertThat(violations)
        .extracting(violation -> violation.getConstraintDescriptor().getAnnotation())
        .allSatisfy(annotation -> assertThat(annotation).isInstanceOf(MaxDataSize.class));
```
With the benefit that you get a better error message on failure

File: src/test/java/io/airlift/units/TestDataSizeValidator.java [hunk 2]
```diff
@@ -79,57 +79,48 @@ public void testAllowsNullMaxAnnotation()
...
+                .allMatch(MaxDataSize.class::isInstance);
+        assertThat(violations.stream().map(violation -> violation.getPropertyPath().toString()))
```
Comment (electrum) [inline src/test/java/io/airlift/units/TestDataSizeValidator.java]:
```java
assertThat(violations)
        .extracting(violation -> violation.getPropertyPath().toString())
        .containsOnly("constrainedByMax", "constrainedByMinAndMax");

File: src/test/java/io/airlift/units/TestDataSizeValidator.java [hunk 3]
```diff
@@ -110,6 +138,13 @@ public void testFailsMaxDataSizeConstraint()
                 .allMatch(MaxDataSize.class::isInstance);
         assertThat(violations.stream().map(violation -> violation.getPropertyPath().toString()))
                 .containsOnly("constrainedByMax", "constrainedByMinAndMax");
+
+        violations = VALIDATOR.validate(new ConstrainedOptionalDataSize(Optional.of(new DataSize(11, MEGABYTE))));
+        assertThat(violations).hasSize(2);
+        assertThat(violations.stream().map(violation -> violation.getConstraintDescriptor().getAnnotation()))
```
Comment (electrum) [inline src/test/java/io/airlift/units/TestDataSizeValidator.java]: Use `extracting` and `allSatisfy`

File: .travis.yml [hunk 4]
```diff
@@ -1,9 +1,8 @@
...
 jdk:
-  - oraclejdk8
-  - oraclejdk9
-  - oraclejdk10
+  - openjdk8
+  - openjdk9
   - openjdk10
   - openjdk11
-  - openjdk-ea
+  - openjdk13
```
Comment (electrum) [inline .travis.yml]: Add `openjdk12` as well?

File: src/test/java/io/airlift/units/TestDataSizeValidator.java (hunk 1 above)
Comment (findepi) [reply src/test/java/io/airlift/units/TestDataSizeValidator.java]: When the final ` .containsOnly(` fails, the exception message is the same. 
But I like the `.extracting`!

File: src/test/java/io/airlift/units/TestDataSizeValidator.java (hunk 3 above)
Comment (findepi) [reply src/test/java/io/airlift/units/TestDataSizeValidator.java]: I don't see `. allSatisfy`. The `.allMatch` here is assertj's not Stream's.

File: .travis.yml (hunk 4 above)
Comment (findepi) [reply .travis.yml]: I thought it's not relevant anymore now, but will add anyway

**Pull request merged**